spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -197,4 +198,87 @@ static Schema getOutputSchema(Schema inputSchema, String predictionField) {
+      if (!(type.equals(Schema.Type.STRING) || (type.equals(Schema.Type.ARRAY) &&
+        componentSchemaType.equals(Schema.Type.STRING)))) {
+        throw new IllegalArgumentException(String.format("Field to be transformed should be of type String or " +
+                                                           "Nullable String or Array of type String or Nullable " +
+                                                           "String . But was %s for field %s.", type, key));
+      }
+      if (value == null) {
+        throw new IllegalArgumentException(String.format("Output column name not specified for column : %s. " +
+                                                           "Please make sure it is in the format 'input-column':" +
+                                                           "'transformed-output-column'.", entry.getKey()));
+      } else if (inputSchema.getField(value) != null) {
+        throw new IllegalArgumentException(
+          String.format("Output column name %s for column %s is already present in the input schema. Please " +
+                          "provide a different output field name.", value, key));
+      }
+    }
+  }
+
+  /**
+   * Get the valid of input field which is to be used for text based feature generation.
+   *
+   * @param input      input Structured Record
+   * @param inputField field to use for feature generation
+   * @param pattern pattern to use for splitting the string
+   * @return
+   */
+  static List<String> getInputFieldValue(StructuredRecord input, String inputField, String pattern) {
+    List<String> text = new ArrayList<>();
+    Schema schema = input.getSchema().getField(inputField).getSchema();
+    if (!schema.isNullable() && input.get(inputField) == null) {
+      throw new IllegalArgumentException(String.format("Null value found for non-nullable field %s", inputField));
+    }
+    Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+    if (type.equals(Schema.Type.ARRAY)) {
+      schema = schema.getComponentSchema();
+      if (!schema.isNullable() && ((ArrayList) input.get(inputField)).isEmpty()) {
+        throw new IllegalArgumentException(String.format("Null value found for non-nullable field %s", inputField));
+      }
+      type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+      if (type.equals(Schema.Type.STRING)) {
+        text = (ArrayList) input.get(inputField);
+      } else {
+        throw new IllegalArgumentException(
+          String.format("Field to be used for text based feaure generation should be of type string, nullable " +
+                          "string or array of strings or nullable string. But was ARRAY of %s for field %s.",
+                        schema.getComponentSchema(), inputField));
+      }
+    } else if (type.equals(Schema.Type.STRING)) {
+      text = Lists.newArrayList(Splitter.on(Pattern.compile(pattern)).split((String) input.get(inputField)));
```

Comment:
utility classes should not maintain any state. Otherwise things get really messy and fail in confusing ways once other code starts calling the method. 

The reason you get that error is that spark is complaining that the plugin is not serializable, because the Splitter is not serializable. Instead, you can just compile the Pattern in initialize and store it, since that is the expensive part.